### PR TITLE
Adds support for new Rekor 'dsse' entry type

### PR DESCRIPTION
**Pull request by bobcallaway**
#### Summary
This add support for uploading Rekor entries about DSSE envelope-based attestations using the new `dsse` type added in https://github.com/sigstore/rekor/pull/1487. 

This PR builds on https://github.com/sigstore/sigstore-js/pull/525 (given the codegen changes can not be cleanly separated).

#### Release Note
- Adds support for uploading entries to Rekor using `dsse` pluggable type

part of #526

### CODE REVIEW

File: packages/client/src/tlog/verify/body.ts [hunk 1]
```diff
@@ -104,6 +132,42 @@ function verifyHashedRekordTLogBody(
...
+  // Turns them into base64-encoded strings for comparison
+  const dsseSigs = dsse.signatures.map((signature) =>
+    signature.sig.toString('base64')
+  );
+
+  // Collect all of the signatures from the tlog entry
+  // Remember that tlog signatures are double base64-encoded
+  const tlogSigs = tlogEntry.spec.signatures?.map((signature) =>
+    signature.signature ? enc.base64Decode(signature.signature) : ''
+  );
```
Comment (bdehamer) [inline packages/client/src/tlog/verify/body.ts:150]: Is this comment about the double base64 encoding still true for the DSSE type?

**Review by bdehamer — COMMENTED** [review]
@bobcallaway I'd like to do try generating/verifying some bundles with the new DSSE Rekor type -- has the build with the DSSE type been deployed to production yet?

Before I merge this, I'm going to add some unit tests around the new verification code as well.

File: packages/client/src/tlog/verify/body.ts (hunk 1 above)
Comment (bobcallaway) [reply packages/client/src/tlog/verify/body.ts:150]: do you have docs somewhere on how to target staging with the client? I had to make several code changes when I was doing that.

Comment (bdehamer) [reply packages/client/src/tlog/verify/body.ts]: There's a new CLI in the "packages/cli" directory which supports a flag for overriding the rekor URL. From the root of the project you can do:

```
./packages/cli/bin/dev attest --help
```

That should show you the various flags

**Pull request merged**